tree/tree/src/TBasket.cxx

```diff
@@ -892,17 +903,33 @@ void TBasket::Streamer(TBuffer &b)
       }
       b << fNevBuf;
       b << fLast;
+      Bool_t mustGenerateOffsets = fEntryOffset && fNevBuf &&
+                                   (fIOBits & static_cast<UChar_t>(TBasket::EIOBits::kGenerateOffsetMap)) &&
+                                   CanGenerateOffsetArray();
+      // We currently believe that in all cases when offsets can be generated, then the
+      // displacement array must be zero.
+      if (mustGenerateOffsets) assert(fDisplacement == nullptr);
```

Comment:
humm ... might be better as just
```
assert( !mustGenerateOffsets || fDisplacement == nullptr );
```

Reply:
Should be the same thing to the compiler... using your suggested syntax nonetheless.